### PR TITLE
Allow alter trusted clone plugins and filter them via tag

### PR DESCRIPTION
**Pull request by 6543**
Make it possible to let an admin define privileged- and trusted-clone-plugins by name **and tag**.

If no tag is specified it will allow all tags as it was before.

### CODE REVIEW

File: cli/exec/exec.go [hunk 1]
```diff
@@ -185,7 +186,10 @@ func execWithAxis(ctx context.Context, c *cli.Command, file, repoPath string, ax
 	}
 
 	// lint the yaml file
-	err = linter.New(linter.WithTrusted(true)).Lint([]*linter.WorkflowConfig{{
+	err = linter.New(
+		linter.WithTrusted(true),
+		linter.WithTrustedClonePlugins(constant.TrustedClonePlugins),
```
Comment (6543) [inline cli/exec/exec.go:191]: INFO: exec normally will just cloning as it execs locally ... but we specify a default to be got to go ...

File: cmd/server/setup.go [hunk 2]
```diff
@@ -165,8 +164,9 @@ func setupEvilGlobals(ctx context.Context, c *cli.Command, s store.Store) error
 	server.Config.Pipeline.AuthenticatePublicRepos = c.Bool("authenticate-public-repos")
 
 	// Cloning
-	server.Config.Pipeline.DefaultCloneImage = c.String("default-clone-image")
-	constant.TrustedCloneImages = append(constant.TrustedCloneImages, server.Config.Pipeline.DefaultCloneImage)
```
Comment (6543) [inline cmd/server/setup.go:169]: we "alter" a _constant_ ... this is bad anyway!!! good it got changed

File: docs/docs/30-administration/10-server-config.md [hunk 3]
```diff
@@ -351,6 +351,16 @@ a user can log into Woodpecker, without re-authentication.
 > Defaults are defined in [shared/constant/constant.go](https://github.com/woodpecker-ci/woodpecker/blob/main/shared/constant/constant.go)
 
 Docker images to run in privileged mode. Only change if you are sure what you do!
+You should specify the tag of your images too, as this enforces exact matches.
```
Comment (6543) [inline docs/docs/30-administration/10-server-config.md:354]:
```suggestion

You should specify the tag of your images too, as this enforces exact matches.
```

File: docs/docs/91-migrations.md [hunk 4]
```diff
@@ -4,6 +4,8 @@ Some versions need some changes to the server configuration or the pipeline conf
 
 ## `next`
 
+- `WOODPECKER_DEFAULT_CLONE_IMAGE` got depricated use `WOODPECKER_DEFAULT_CLONE_PLUGIN`
+- Check trusted-clone-plugins by image name and tag (if tag is set)
```
Comment (6543) [inline docs/docs/91-migrations.md:8]: I'm not sure if we should add it to migrations, as normally an admin dont have to do anything even if he specified an image ...

**Review by lafriks — APPROVED** [review]

File: pipeline/frontend/yaml/linter/linter_test.go [hunk 5]
```diff
@@ -173,6 +173,10 @@ func TestLintErrors(t *testing.T) {
 			from: "{steps: { build: { image: plugins/docker, settings: { test: 'true' } } }, when: { branch: main, event: push } } }",
 			want: "Cannot use once privileged plugins removed from WOODPECKER_ESCALATE, use 'woodpeckerci/plugin-docker-buildx' instead",
 		},
+		{
+			from: "{steps: { build: { image: golang, settings: { test: 'true' } } }, when: { branch: main, event: push }, clone: { git: { image: woodpeckerci/plugin-git:v1.1.0 } } }",
```
Comment (6543) [inline pipeline/frontend/yaml/linter/linter_test.go:177]:
```suggestion
			from: "{steps: { build: { image: golang, settings: { test: 'true' } } }, when: { branch: main, event: push }, clone: { git: { image: some-other/plugin-git:v1.1.0 } } }",
```

File: pipeline/frontend/yaml/linter/option.go [hunk 6]
```diff
@@ -30,3 +30,10 @@ func PrivilegedPlugins(plugins []string) Option {
 		linter.privilegedPlugins = &plugins
 	}
 }
+
+// PrivilegedPlugins adds the list of privileged plugins.
```
Comment (6543) [inline pipeline/frontend/yaml/linter/option.go:34]:
```suggestion
// WithTrustedClonePlugins adds the list of trusted clone plugins.
```

**Pull request merged**